### PR TITLE
Release preparations for v0.23.0

### PR DESCRIPTION
**Pull request by ansiwen**
Fixes: #892

### CODE REVIEW

**Comment by ansiwen** [comment]
I tried to use "promote" mode of apiage.py, but I couldn't get it to work, maybe I misunderstand the feature. So I did the move from preview to stable in `api-status.json` manually. I would like to add a remark in the release docs, so I will not try that again in  couple of months. So what is the right way?

**Comment by phlogistonjohn** [comment]
I think you may have forgotten that the script cross checks the state of the Go files with the JSON. You edited the JSON only. The tool expects you to edit the Go files to fix the ceph_preview flags and then you can run `make api-promote` to automate the updating of the JSON file.

The tests are all failing because the JSON file now disagrees with the Go sources. :-)

IOW my workflow would roughly be:
* `make pi-report-updates  RESULTS_DIR=$PWD/_results
* `cat _results/updates-found.md`
* for each dir in report
  * $EDITOR $dir/<...files...>  - remove ceph_preview build tags
  * `make api-promote  RESULTS_DIR=$PWD/_results`  - update json
  * `git commit -sp`  - commit changes

I do like to see at least one commit per directory (e.g. `rbd`, `cephfs`, `cephfs/admin`, and so on...).  

If you'd prefer I add this to the release docs, I'm happy to do it. But I'll let you run through the process to validate it in case I missed anything.

**Comment by phlogistonjohn** [comment]
README change looks good BTW.

**Comment by ansiwen** [comment]
Doh. 🤦‍♂️ I completely forgot about the preview tag in the code.

**Review by phlogistonjohn — APPROVED** [review]
LGTM

File: cephfs/makedirs_test.go [hunk 1]
```diff
@@ -14,7 +11,7 @@ func TestMakeDirs(t *testing.T) {
 	defer fsDisconnect(t, mount)
 
 	dir1 := "/base/sub/way"
-	err := mount.MakeDirs(dir1, 0755)
+	err := mount.MakeDirs(dir1, 0o755)
```
[the remark this answers is not in the document]
Comment (ansiwen) [reply cephfs/makedirs_test.go:14]: Yes, my IDE runs gofmt when I save. So maybe that changed?

[the remark this answers is not in the document]
Comment (phlogistonjohn) [reply cephfs/makedirs_test.go:14]: Seem like that's the case :-)

**Review by anoopcs9 — APPROVED** [review]

**Pull request merged**